app/models/redirection.rb

```diff
@@ -0,0 +1,16 @@
+class Redirection < ActiveRecord::Base
+  belongs_to :next, class_name: "Redirection"
+  has_one :previous, class_name: "Redirection", foreign_key: "next_id"
+
+  validates :next_id, uniqueness: true, allow_blank: true
```

Comment:
Oh, I think this was a concession to adding the first site. How can you have a next when there is no next to go to? But I think we can have next be itself. That should validate?